### PR TITLE
Working version1

### PR DESCRIPTION
**Pull request by diego-hdz1**
- Consuming backend data dynamically in the frontend. (Added Pop up Modal when clicking to show more info)
- Fixed some comments from the previous PR

### CODE REVIEW

File: Amadeus-frontend/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,19 @@
 import './App.css'
-// import Search from './components/search'
+import { BrowserRouter, Routes, Route, useLocation } from "react-router-dom";
+import ShowResults from './components/showResults'
 import Search from './components/search'
 
 function App() {
+  //const location = useLocation();
```
Comment (mayralgr) [inline Amadeus-frontend/src/App.tsx:7]: is this useful?

File: Amadeus-frontend/src/components/search.tsx [hunk 2]
```diff
@@ -5,18 +5,18 @@ import dayjs from 'dayjs';
 import customParseFormat from 'dayjs/plugin/customParseFormat';
 
 type FieldType = {
-  username?: string;
-  password?: string;
+  departureAirport?: string;
```
Comment (mayralgr) [inline Amadeus-frontend/src/components/search.tsx:8]: is all of this truly optional?

File: Amadeus-frontend/src/components/search.tsx [hunk 3]
```diff
@@ -71,9 +80,8 @@ const Search: React.FC = () => (
       rules={[{ required: true, message: 'Please a date' }]}
     >
     <DatePicker
-    defaultValue={dayjs('2019-09-03', dateFormat)}
-    minDate={dayjs('2019-08-01', dateFormat)}
-    // maxDate={dayjs('2020-10-31', dateFormat)}
+    minDate={today}
+    //TODO: It needs to change dynamically
```
Comment (mayralgr) [inline Amadeus-frontend/src/components/search.tsx:84]: Why is nor dynamic right now?

File: Amadeus-frontend/src/components/showResults.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        setIsModalOpen(true);
+      };
+
+    const handleCancel = () => {
+        setIsModalOpen(false);
+        setCurrentCard(null);
+      };
+
+    const fetchData = () => {
+        const url = `http://localhost:8080/flights?departureAirportCode=SYD&arrivalAirportCode=BKK&departureDate=2025-05-02&returnDate=2025-05-05&adults=1&nonStop=false&currencyCode=EUR`;
```
Comment (mayralgr) [inline Amadeus-frontend/src/components/showResults.tsx:24]: centralize this url

File: AmadeusBackend/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+I will add the steps to set up the .env files here
```
Comment (mayralgr) [inline AmadeusBackend/README.md:1]: Try to add them in this pr

File: AmadeusBackend/src/main/java/com/Encora/AmadeusBackend/Service/FlightService.java [hunk 6]
```diff
@@ -17,9 +16,6 @@ public interface FlightService {
     String getAirportName(String airportCode);
     String getCityName(String airportCode);
     String getTotalTime(String initialTime, String finalTime);
-//    List<FlightDetails> getFlightDetails();
-    //METODO PARA CHECAR TODO, VERIFICAR Y LANZAR EXCEPCIONES
-
-
+    //Here I need to implement a validation method
```
Comment (mayralgr) [inline AmadeusBackend/src/main/java/com/Encora/AmadeusBackend/Service/FlightService.java:19]: add todos

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    image: amadeus-frontend-image
+    container_name: frontend
+    ports:
+      - "3000:3000"
+    depends_on:
+      - backend
+    networks:
+      - app-network
+
+networks:
```
Comment (mayralgr) [inline docker-compose.yaml:22]: why you need networks?

**Review by mayralgr — CHANGES_REQUESTED** [review]

File: Amadeus-frontend/src/components/search.tsx (hunk 3 above)
Comment (diego-hdz1) [reply Amadeus-frontend/src/components/search.tsx:84]: Because I need to test if the minimum date is todays date + 1, and I thing that validation its not always dynamic

File: docker-compose.yaml (hunk 7 above)
Comment (diego-hdz1) [reply docker-compose.yaml:22]: I don't need them to successfully communicate the containers between them? I also read that they are a good practice for isolate the services, but maybe I misunderstood something

File: Amadeus-frontend/src/components/search.tsx (hunk 3 above)
Comment (mayralgr) [reply Amadeus-frontend/src/components/search.tsx:84]: what do you mean validation is not always dynamic

File: docker-compose.yaml (hunk 7 above)
Comment (mayralgr) [reply docker-compose.yaml:22]: there is a default network created https://docs.docker.com/compose/how-tos/networking/ please review the documentation
